### PR TITLE
Add `enableDebugWireframe` to use wireframes for ModelExperimental in WebGL1

### LINKED ISSUE #10340: Better handling of `ModelExperimental` indices in WebGL1
This came up with #10332 being merged.

WebGL1 doesn't have the ability to get data from a buffer on the GPU. Therefore, the index buffer must be stored in CPU memory to compute new indices for the wireframe. However, this is unideal for users who won't ever use `debugWireframe` for their purposes.

There are two possible solutions:

1. **Find a way to store the index buffer's resource loader, and load the index buffer originally on the GPU. If `debugWireframe` is enabled, then reload this resource as a typed array.**
  a. This might complicated to figure out, but it could be possible. Could we store the index resource loader separately from the others in `GltfLoader`?
  b. This would also only involve changes to the loading system; there'd be no changes to the public API.
2. **Add a `debug` option or flag to `ModelExperimental` that the user must specify before the model is loaded. If this flag is true, then load the index buffer as a typed array.**
  a. This would simplify the loading logic. It may also be expandable if other debug options need this in the future (though I'm not sure what those would be?)
  b. This would involve changes to the API for both models and tilesets. It also may not be intuitive that having `debug: true` is necessary for `debugWireframe` to work, but not for `debugShowBoundingVolume`.

### PR DESCRIPTION
**Pull request by j9liu**
Closes #10340. Currently wireframes in `ModelExperimental` require the indices to be loaded as typed arrays for WebGL1. However, this will result in performance costs for users that don't need the `debugWireframe` option.

The preferred solution is to require an option, `enableDebugWireframe`, to be explicitly set as `true` in the `ModelExperimental` constructors. If this is true, then `debugWireframe` will work for models in WebGL1. However, if this is not true, then toggling `debugWireframe` will have no effect. This will not affect models in WebGL2 because in WebGL2, data can be pulled from GPU buffers.

[`ModelExperimental` Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=tVdtb9s2EP4rhPehzmDLsuMkrpcWy1uzDg0a1GkLbB4GSjpbhClSIynbWZH/viMpyXLsbF2LOLEtkkfec3cP786xFNqQJYMVKPKKCFiRC9CsyIJPbq49bcVufCGFoUyAmrY65MtUEKJTWfDkTLCMGhgTowro2PkYJWFt3ueG4eFjL0xIr0c+ilhmGQhDjCQGULFJqSErpmCmaAaarKRakJlU5DNE1+8G1U4FfxUojpNzPvCq7NLDVDwc/DQVU+ENCGIu40UQF0qhkjuWAZpUmvNrwRkVlwg1mCmZvdVydBz22/aYaWsQDvrdfr8bvrwbhOPByfjwJAj7x+HxcHhydDgcHY+GLw9Pfpu2psLrQ2NuZAJcT9Ek68HMjVDd7/bE0mTrhjEe7yGQc8q5lAId6JdxI3CIUWRWiNh6i7QPqr3oX7fo1LSrOQs2CHr4P6FZzgHtoT2vu+e1lEq2R8GcR7Va++ofVc/WHvv94Fb95yP4kwUTAhJykVJFY+Mo8IwW3NAK/Z/4uAM9/B/ILxGvJBcyyxVojSY4LM8F32nbKKvMYXxxpwrk5Zyb2Teb8lFwZsi5XH8T+icx44Hu5PqhQknCr/KvFPC8bHAqms/fx+VrJQuRkE+Qspg/G3SvpVSyPfouPr/FVENFjET+ViJ8BXg8utazNdgl8L+5/g+XKOO6vjgAmCAdrgSiYv65yvtXgkYcEsRPubZF5KGxN80fFaZfgCZMzG+ZidMPknujyrUbatLAyA8oQYVu90fhgQMU+k+fvv25M7aG5I1Vf6dQFotOtqkW9ZQOsKRQ7uXkm3rPNWAppEaqsoIIqUxa+mbaWmGx2hQLDmV9wFEdoWZUCsU7JAU2T00ZNw8xl5o54RrXBVVYN9G2Q1fFLmGuAHQZ1m5/cBiEJ8Phcf9lGaThMAiPwsOT8Lic8Frss8dGytof6BgtCnLFMlS5BB0oyOQSztC/PrZZGb+n5GmSlDiqWNgNV+scUAbLMeUO8jVyqF1z0zJqTKz91YxTg2FUbD3eE430UfCbMWmwu/Jcg6vIo8aoPPmKc5ZryZLg8/VkNGwI7KFHzfUNWIbdT5HdoiyfsL+xC+oPRptVurarE+QPrgxCfNVr2EBcVL1QyjSRxdNNEVVAEqbdHSFM+N6oX50E7vJcbl2oRjdmX8mjxfo6BnvvYXmTD5oscWFBTtDk/hajyDQEJgXR3mQcJ1Gnnaohw1MVxTskF2emjs9OdJ663VTMm1Hdf8mHRwed/5Lp9reEvDWRTcuoa5KnoCBQKFto8iPZJDX/UPmg2oc9EDLeN762xbXUt/ekZjWam4+3rkEt/A6Xgg9Xt1dnd53Kz+56ua8Hq2hCRRJTbbBW4MF3UvKIqhsQhXexrjtQJGa8IChNYgyMASQOkIg5m7Trou2EjYQHXmU+44/Eu5zIuLAcDOZgrjjYx/P7t3iPW6XMtGW1lZYsBHbXSNTAWMXtmkX7RGie8/vzEstGtFMp37PHesECeR9pUEvLxOa+aWs/V1suSIEuIh0rFkGDkUvKC6gZme0hvE1nVmgTAfvX6rROtbnn8LoK6M8syzHB20TVxpppAGsmOlz3IuzrwASx1lUBJOSHyr9fNoSL0GFz1waMiZpHtD0cdEj1DoPRZjfeDow6em1Mhvm6MR1JlYDqepZuLz7sqGYix4TSALAEZRgmoi7lbC7GmLmShMOu1q6x1B1saa6WImmMzLZXd1W7FA1b1s/wF2F35SrPGO3gyePtp72mv08TtiQsebXnpyeJOdUaV2YFdxl32np92kP5na1cujzyHg3n9N6Kpf3X7/xkEASnPRzu31lTvw7/qXenuc/BgkohXkS2//LksWhs34LjBNuorr2BlZhtavbytuP3fswT9+P5hdPwoqFzksrVpghUbtrg/Qc) and [3D Tiles sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=zVh7Uxs3EP8qqicTzhNzBidMWtswLYYktBCY4CZ/YKaR74RPQZauks7gMP7uXUn30Dkm4ZWZZphYt9rHb1d7q90bEEWzabh/nRJJp4RrzN4QrDNJVEg4HjNyJGLC/H20jbTMSG/EI8GVRjNKrogEKidXaOD0fbS0YNSI7PNAgCDlRI4aLXQz4gipBMfiSnWtqpahgDJNrvVxqimo7To2hNpt9DcYmhrbSAukCZjUCdboikpyIfGUKHQl5CW6EBJ9IuO3h51CUpJ/M2AH4oR1nCmztRjxRRPgO+AhGCU8DnLkjugeXu4NKSPqgKuURFrII3pNebN0nBZ0460NEwQhVxqtVGC1W05QUgKImIguwyiTEnwcQpTrsfwzYxTzPaxJ0Hm19Xqr0wk3t169/q3TtEocFhURTirz9hG2GYFgGQhEe8wzD68Nc84CUT9zscuDjxCH+HbRqDF0HHB+xQ5kiMhkRLoFAQFbGLbh7xRPU0YAMW7XAtnOtZSLdm44/KIEL3UvWreBkJgrloFvjwWyi3WUkLj49RQ/HFL7OMWQbz8NmVMPOXhPiEcZ03Q9EkzIp8U2MCrVPdF8cO8k+ihYNiVPnE+fqE5c6cnNOCv3hJi797TBMtDscmiK6j0RQQHRmEePxlTqqVZPg6t9LKm5HEztfnqMnvJ7AhyIaSoU1Y9NtFKPt7ofkhNBuR4wkT32DCtF3vLD290H4xkIG2c9eIIKsRJczcCDYb4XcoqZ+hkAc9UPhvY0BWMltKKKPBTansSR+BnArOLvwTq3/yvCoPOBe8wxdlHGY3IBrWDcWtUGLmyXkjc+lxz6IpHpUIOty6DsWvy+RwvBxth0n7GIMtMkhhOi9xkxy935QQw9aM4zahjBZd04Tdl8l/KY8omqbLQKzc1ViAx0Y+d4rIicmcLpS44auVtg0XKqbKwiScckuMh4ZMoYClxzHTeLwysaNycJ/uQMPduxPgxFPfbfQyNc112ioReoaIjz4wryo242q3SzXWaYwmBANZ3BxCDJVMxIydrL86HS+cuS0sKup/TbrjrM9UFQyuzpVQkNwwqvLLlFJeE3xD5UHENqFK9N1WzXMp/o4KZ6TzLJuigHHBbvUavah0llUMwpCVUITujWgQVLgmKq7AmDx25u2ax0ucPfI+Ns8qmQ8mYl62vTrVxqeE4DNhzPT6SYUkUK7lAnhHsH7gX8ByGvxoeCPQ/pVyGmQxFU9BJBy6d5wX0HwOA1O6FQ0z5gPiE1YYQ2WvXn9U64TDrCOgmn+DrY3NgIN9B66fRYZPYdPk0TIkkowVKmWgiYmr4G76EKnPnnykkqBQy6mhJVeR5WRC8IJpt9xUuJXck00fPnd2KE8NBJon2E/iFV56v0nC3NiH7anprt4KYeN9v+d9ES1bod03ziPlveREBqPLtxuBZoZxv9+tJM8DDWG3XBWppJuDLWTJi3mqPGeesOGjZqGiSJ1+4o+bouKaTJoLsKb3ZqwnPCmLi6s+WaLIMJ/c5ma5LRHPPvSNrvEz7/mGW3WFqmLGrPi6aXqIvlggHFIDJdhX8ZSSlkLdt0IsUVsvRepcC/jMqrJly6Z/KiW6sj6mzj3Lu1E8xjtvzN6DSShPDTFEdkfwZV9J1jCoqvJJjPsHK2c3mwrA94muk/rB+eQ+YaMpU498lZvXAftsDqioKXb1oPV9x++XZ1UdXqxfy9rerbhQlzK5/4W0FxIk6MET7RCfDX5ENHzjnN56zAfL+hwLfRg59+LtdDL15Q77S+hbKs+Yye93xuwUjIxMQrYJ+f3fgSiy56drPCmcBnai4+FwpqV735vta67VSH85SEh/tvhv8MDg8Gf/1vzrNwViWQ93CQMA844R97c3Swt3e4X/nTaDX6tkbvFOH5ncLQKLVpIQLotDWBThtDY9AeZ9ElVPRIqSKC/bYv2o/pDNF4e8WHVBQxrBTsXGSMndKvUDp2+m3g/0aUCXvtHs+IZHhu2JLNnUNHDMOw34bH1ZJl51x60ncvO4phTFgfQ9sMXKL4ZFu87K2iTRqSa2j518xYApfEDEM96y6PBCXzANtf4B8kQiiCMMpZAOSa9c6JVmAM5j41GYM0nIQJU0Kiy7G4hvj4EC2ZxF1UduY76NQta5576/8A) for testing. I left comments for the lines that should be toggled on / off for testing.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/debug-wireframe-webgl1/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Specs/Scene/GltfIndexBufferLoaderSpec.js [hunk 1]
```diff
@@ -269,6 +267,7 @@ describe(
 
     afterAll(function () {
       scene.destroyForSpecs();
+      sceneWithWebgl2.destroyForSpecs();
```
Comment (IanLilleyT) [inline Specs/Scene/GltfIndexBufferLoaderSpec.js:270]: nice catch!

File: Source/Scene/ModelExperimental/buildDrawCommands.js [hunk 2]
```diff
@@ -218,17 +225,17 @@ function createWireframeIndexBuffer(primitiveRenderResources, frameState) {
...
-    if (useWebgl2 && defined(indicesBuffer)) {
-      originalIndices = IndexDatatype.createTypedArray(
-        vertexCount,
-        indicesCount
-      );
+    if (defined(indicesBuffer)) {
+      const useUint8Array = indicesBuffer.sizeInBytes === indicesCount;
+      originalIndices = useUint8Array
+        ? new Uint8Array(indicesCount)
+        : IndexDatatype.createTypedArray(vertexCount, indicesCount);
```
Comment (IanLilleyT) [inline Source/Scene/ModelExperimental/buildDrawCommands.js:237]: @lilleyse do you know why `IndexDatatype.createTypedArray` doesn't output 8-bit indices, and if there's any code that depends on it outputting 16/32 only? Or if there's any code that also has this workaround.

**Review by IanLilleyT — CHANGES_REQUESTED** [review]

**Comment by IanLilleyT** [comment]
I turned wireframe on and off (both before and after this PR) in WebGL1 in the [Model Experimental 3D Model Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html?src=Model%20Experimental%203D%20Models.html) and the memory reduction is working as expected!

**New + No Wireframe:**

Screen Shot 2022-05-03 at 2 04 52 PM

**New + Wireframe:**

Screen Shot 2022-05-03 at 2 04 59 PM

**Old + No Wireframe:**

Screen Shot 2022-05-03 at 2 08 33 PM

**Old + Wireframe:**

Screen Shot 2022-05-03 at 2 08 03 PM

**Review by IanLilleyT — APPROVED** [review]

**Comment by IanLilleyT** [comment]
Thanks @j9liu !

**Pull request merged**